server/sonar-server/src/main/java/org/sonar/server/component/ws/TreeAction.java

```diff
@@ -334,4 +334,143 @@ private static TreeWsRequest toTreeWsRequest(Request request) {
     return ordering.nullsLast().onResultOf(function);
   }
 
+  private static class Request {
+    private String baseComponentId;
+    private String baseComponentKey;
+    private String component;
+    private String branch;
+    private String strategy;
+    private List<String> qualifiers;
+    private String query;
+    private List<String> sort;
+    private Boolean asc;
+    private Integer page;
+    private Integer pageSize;
+
+    /**
+     * @deprecated since 6.4, please use {@link #getComponent()} instead
+     */
+    @Deprecated
+    @CheckForNull
+    private String getBaseComponentId() {
+      return baseComponentId;
+    }
+
+    /**
+     * @deprecated since 6.4, please use {@link #setComponent(String)} instead
+     */
+    @Deprecated
+    private Request setBaseComponentId(@Nullable String baseComponentId) {
+      this.baseComponentId = baseComponentId;
+      return this;
+    }
+
+    /**
+     * @deprecated since 6.4, please use {@link #getComponent()} instead
+     */
+    @Deprecated
+    @CheckForNull
+    private String getBaseComponentKey() {
+      return baseComponentKey;
+    }
+
+    /**
+     * @deprecated since 6.4, please use {@link #setComponent(String)} instead
+     */
+    @Deprecated
+    private Request setBaseComponentKey(@Nullable String baseComponentKey) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)